File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/PolylineController.java
package com.valentingrigorean.arcgis_maps_flutter.map;

import android.graphics.Color;

import com.esri.arcgisruntime.mapping.view.Graphic;
import com.esri.arcgisruntime.symbology.SimpleLineSymbol;

public final class PolylineController extends BaseGraphicController {
    private final Graphic graphic;
    private final SimpleLineSymbol polylineSymbol;

    public PolylineController(String polylineId) {
        polylineSymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 10);
        graphic = new Graphic();
        graphic.setSymbol(polylineSymbol);
        graphic.getAttributes().put("polylineId", polylineId);
    }

    @Override
    protected Graphic getGraphic() {
        return graphic;
    }

    public void setColor(int color) {
        polylineSymbol.setColor(color);
    }

    public void setWidth(float width) {
        polylineSymbol.setWidth(width);
    }

    public void setStyle(SimpleLineSymbol.Style style) {
        polylineSymbol.setStyle(style);
    }

    public void setAntialias(boolean antialias) {
        polylineSymbol.setAntiAlias(antialias);
    }

}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/PolylineController.kt
package com.valentingrigorean.arcgis_maps_flutter.map

import android.graphics.Color
import com.esri.arcgisruntime.mapping.view.Graphic
import com.esri.arcgisruntime.symbology.SimpleLineSymbol

class PolylineController(polylineId: String?) : BaseGraphicController(), PolylineControllerSink {
    protected override val graphic: Graphic
    private val polylineSymbol: SimpleLineSymbol

    init {
        polylineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.BLACK, 10f)
        graphic = Graphic()
        graphic.symbol = polylineSymbol
        graphic.attributes["polylineId"] = polylineId
    }

    override fun setColor(color: Int) {
        polylineSymbol.color = color
    }

    override fun setWidth(width: Float) {
        polylineSymbol.width = width
    }

    override fun setStyle(style: SimpleLineSymbol.Style?) {
        polylineSymbol.style = style
    }

    override fun setAntialias(antialias: Boolean) {
        polylineSymbol.isAntiAlias = antialias
    }
}